### PR TITLE
Added capability to run example payment channels on regtest, test or main nets

### PR DESCRIPTION
**Pull request by w-shackleton**
I added an extra command line option to the `ExamplePaymentChannelClient` and `ExamplePaymentChannelServer` programs to allow them to run on testnet or mainnet as well as regtest net.

### CODE REVIEW

File: examples/src/main/java/org/bitcoinj/examples/ExamplePaymentChannelClient.java [hunk 1]
```diff
@@ -54,8 +56,22 @@
 
     public static void main(String[] args) throws Exception {
         BriefLogFormatter.init();
-        System.out.println("USAGE: host");
-        new ExamplePaymentChannelClient().run(args[0]);
+        System.out.println("USAGE: [--net={MAIN/TEST/REGTEST}] host");
```
Comment (schildbach) [inline examples/src/main/java/org/bitcoinj/examples/ExamplePaymentChannelClient.java]: In several tools, we're using "jopt" for argument parsing. Perhaps you want to introduce it here as well?

File: examples/src/main/java/org/bitcoinj/examples/ExamplePaymentChannelClient.java [hunk 2]
```diff
@@ -79,7 +95,11 @@ public void run(final String host) throws Exception {
                 return ImmutableList.<WalletExtension>of(new StoredPaymentChannelClientStates(null));
             }
         };
-        appKit.connectToLocalHost();
+        // Broadcasting can take a bit of time so we up the timeout for "real" networks
+        final int timeoutSeconds = params == RegTestParams.get() ? 15 : 150;
```
Comment (schildbach) [inline examples/src/main/java/org/bitcoinj/examples/ExamplePaymentChannelClient.java]: Unfortunately, RegTestParams.get() has side effects. If you want to test for specific params, can you use NetworkParams.getId()?

**Comment by albrechta** [comment]
I cannot start the server due to a nullpointer exception (see stacktrace below). Note that there does not exist a wallet yet in my example folder (i.e. no payment_channel_example_server.wallet). 

Are you aware of this issue?

I tried to pass the wallet as an argument to the constructor of `StoredPaymentChannelServerStates` as follows:
```
 appKit = new WalletAppKit(params, new File("."), "payment_channel_example_server") {
            @Override
            protected List<WalletExtension> provideWalletExtensions() {
                return ImmutableList.<WalletExtension>of(new StoredPaymentChannelServerStates(wallet()));
            }
        };
```
However, this does not fix the issue because `wallet()` still returns `null` during the serialization procedure triggered by `createOrLoadWallet`.

Stacktrace:
```
15:13:03 11 WalletAppKit.startUp: Starting up with directory = .
15:13:03 11 MnemonicCode.toSeed: PBKDF2 took 310.5 ms
15:13:03 11 KeyChainGroup.addAndActivateHDChain: Creating and activating a new HD chain: org.bitcoinj.wallet.DeterministicKeyChain@353848bd
15:13:03 11 DeterministicKeyChain.maybeLookAhead: 1 keys needed for M/0H/0 = 1 issued + 0 lookahead size + 0 lookahead threshold - 0 num children
15:13:03 11 DeterministicKeyChain.maybeLookAhead: Took 1.388 ms
15:13:03 11 Wallet.saveToFile: Failed whilst saving wallet
java.lang.NullPointerException
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.getNetworkParameters(StoredPaymentChannelServerStates.java:293)
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.serializeWalletExtension(StoredPaymentChannelServerStates.java:216)
	at org.bitcoinj.store.WalletProtobufSerializer.populateExtensions(WalletProtobufSerializer.java:230)
	at org.bitcoinj.store.WalletProtobufSerializer.walletToProto(WalletProtobufSerializer.java:202)
	at org.bitcoinj.store.WalletProtobufSerializer.writeWallet(WalletProtobufSerializer.java:127)
	at org.bitcoinj.core.Wallet.saveToFileStream(Wallet.java:1446)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1260)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1300)
	at org.bitcoinj.kits.WalletAppKit.createOrLoadWallet(WalletAppKit.java:377)
	at org.bitcoinj.kits.WalletAppKit.startUp(WalletAppKit.java:275)
	at com.google.common.util.concurrent.AbstractIdleService$2$1.run(AbstractIdleService.java:54)
	at com.google.common.util.concurrent.Callables$3.run(Callables.java:95)
	at java.lang.Thread.run(Thread.java:745)
15:13:03 11 Wallet.saveToFile: Temp file still exists after failed save.
Exception in thread "main" Exception in thread " STARTING" java.lang.IllegalStateException: Expected the service to be RUNNING, but the service has FAILED
	at com.google.common.util.concurrent.AbstractService.checkCurrentState(AbstractService.java:285)
	at com.google.common.util.concurrent.AbstractService.awaitRunning(AbstractService.java:229)
	at com.google.common.util.concurrent.AbstractIdleService.awaitRunning(AbstractIdleService.java:151)
	at org.bitcoinj.examples.ExamplePaymentChannelServer.run(ExamplePaymentChannelServer.java:87)
	at org.bitcoinj.examples.ExamplePaymentChannelServer.main(ExamplePaymentChannelServer.java:64)
Caused by: java.lang.NullPointerException
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.getNetworkParameters(StoredPaymentChannelServerStates.java:293)
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.serializeWalletExtension(StoredPaymentChannelServerStates.java:216)
	at org.bitcoinj.store.WalletProtobufSerializer.populateExtensions(WalletProtobufSerializer.java:230)
	at org.bitcoinj.store.WalletProtobufSerializer.walletToProto(WalletProtobufSerializer.java:202)
	at org.bitcoinj.store.WalletProtobufSerializer.writeWallet(WalletProtobufSerializer.java:127)
	at org.bitcoinj.core.Wallet.saveToFileStream(Wallet.java:1446)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1260)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1300)
	at org.bitcoinj.kits.WalletAppKit.createOrLoadWallet(WalletAppKit.java:377)
	at org.bitcoinj.kits.WalletAppKit.startUp(WalletAppKit.java:275)
	at com.google.common.util.concurrent.AbstractIdleService$2$1.run(AbstractIdleService.java:54)
	at com.google.common.util.concurrent.Callables$3.run(Callables.java:95)
	at java.lang.Thread.run(Thread.java:745)
java.lang.NullPointerException
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.getNetworkParameters(StoredPaymentChannelServerStates.java:293)
	at org.bitcoinj.protocols.channels.StoredPaymentChannelServerStates.serializeWalletExtension(StoredPaymentChannelServerStates.java:216)
	at org.bitcoinj.store.WalletProtobufSerializer.populateExtensions(WalletProtobufSerializer.java:230)
	at org.bitcoinj.store.WalletProtobufSerializer.walletToProto(WalletProtobufSerializer.java:202)
	at org.bitcoinj.store.WalletProtobufSerializer.writeWallet(WalletProtobufSerializer.java:127)
	at org.bitcoinj.core.Wallet.saveToFileStream(Wallet.java:1446)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1260)
	at org.bitcoinj.core.Wallet.saveToFile(Wallet.java:1300)
	at org.bitcoinj.kits.WalletAppKit.createOrLoadWallet(WalletAppKit.java:377)
	at org.bitcoinj.kits.WalletAppKit.startUp(WalletAppKit.java:275)
	at com.google.common.util.concurrent.AbstractIdleService$2$1.run(AbstractIdleService.java:54)
	at com.google.common.util.concurrent.Callables$3.run(Callables.java:95)
	at java.lang.Thread.run(Thread.java:745)

```

File: examples/src/main/java/org/bitcoinj/examples/NetworkEnum.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package org.bitcoinj.examples;
```
Comment (schildbach) [inline examples/src/main/java/org/bitcoinj/examples/NetworkEnum.java]: This new file is missing the license header.

**Comment by albrechta** [comment]
I did some more digging regarding the issue outlined above. It is not directly related to this pull request but rather introduced by another commit a few weeks ago. I will open a separate issue such that this PR can be merged (it is quite useful to me, thank you for your work w-shackleton).

**Comment by schildbach** [comment]
Merged.

@albrechta, yes, feel free to open a PR if you find and fix an issue.

**Pull request closed without merging**